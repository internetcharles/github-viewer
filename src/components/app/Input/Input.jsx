import React from 'react';
import { useDispatch, useSelector } from 'react-redux';
import { fetchUser, setSearch } from '../../../actions/userActions';

const Input = () => {
  const dispatch = useDispatch();
  const username = useSelector(state => state.search);

  const handleSubmit = (e) => {
    e.preventDefault();
    dispatch(fetchUser(username));
  };

  const handleChange = ({ target }) => {
    dispatch(setSearch(target.value));
  };

  return (
    <form onSubmit={handleSubmit}>
      <input placeholder="GitHub Username" onChange={handleChange} />
      <button>Submit</button>
    </form>
  );
};

export default Input;
