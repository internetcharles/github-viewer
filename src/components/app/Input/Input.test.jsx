import React from 'react';
import { render, cleanup } from '@testing-library/react';
import Input from './Input';
import { Provider } from 'react-redux';
import store from '../../../store';

describe('Input component', () => {
  afterEach(() => cleanup());
  it('renders Input', () => {
    const { asFragment } = render(
      <Provider store={store}>
        <Input />
      </Provider>
    );
    expect(asFragment()).toMatchSnapshot();
  });
});
