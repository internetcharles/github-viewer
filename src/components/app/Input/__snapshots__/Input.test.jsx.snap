// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Input component renders Input 1`] = `
<DocumentFragment>
  <form>
    <input
      placeholder="GitHub Username"
    />
    <button>
      Submit
    </button>
  </form>
</DocumentFragment>
`;
